src/batched/KokkosBatched_Gemm_Serial_Impl.hpp

```diff
@@ -400,6 +435,52 @@ struct BatchedSerialGemmFunctor {
     return subview_wrapper(v, i1, i3, i2, layout_tag);
   }
 
+  void run() {
+    using execution_space = typename CViewType::device_type::execution_space;
+    using policy_type =
+        Kokkos::RangePolicy<ArgResultsPerThread, execution_space>;
+    Kokkos::parallel_for("BatchedSerialGemm", policy_type(0, batch_size),
+                         *this);
+  }
+
+ public:
+  int invoke() {
+    if (std::is_same<ArgResultsPerThread, ResultsPerThread::Rank0>::value) {
```

Comment:
When ```ResultsPerThread::Rank0``` is specified, it means the algorithm is ```KK_SERIAL_OPT2```. So, ```KK_SERIAL_OPT2``` can be called with ```KK_SERIAL```. Am I correct?

Reply:
> When `ResultsPerThread::Rank0` is specified, it means the algorithm is `KK_SERIAL_OPT2`.

That's right.


> So, `KK_SERIAL_OPT2` can be called with `KK_SERIAL`. Am I correct?

No, the external `BatchedGemm` interface does not support specifying results per thread. You could call the internal `BatchedSerialGemm` interface which is what the `BatchedGemm` interface does.

Reply:
In the ```BatchedGemm```, it selects whether to calculate a rank-0 or rank-2 result per thread in https://github.com/e10harvey/kokkos-kernels/blob/473099cefc13dddfdddb40f2d21da5b21f9e9abf/src/batched/KokkosBatched_Gemm_Decl.hpp#L340-L342
which means ```KK_SERIAL_OPT2``` can be called even when ```KK_SERIAL``` is specified?

Reply:
Gotcha. Thanks for catching this. I will refactor this code so that these template selections are only applied to the SQUARE algorithm type.